### PR TITLE
add conform flag to parametric_view

### PR DESCRIPTION
**Pull request by j042**
# Overview
add conform flag to parametric_view. then resulting spline has conforming parametric space and cps are placed on parametric space "box". Expected use case is in interactive mode, where you can set boundary conditions based on difference in control point locations in parametric view. Added test
Also fixes some errors in create

## Addressed issues
*  wrong scaling in `from_bounds()`

## Checklists
* [x] Documentations are up-to-date.
* [ ] Added example(s)
* [x] Added test(s)

### CODE REVIEW

**Comment by jzwar** [comment]
This PR looks reasonable to me for non-rational splines. However I am not sure about the meaningfulness of the parametric view of rational splines. For example, consider the following spline:
```
import splinepy as sp
import numpy as np
# Create a nurbs
nurbs = sp.NURBS(
    degrees=[2, 1],
    knot_vectors=[
        [0.0, 0.0, 0.0, 0.5, 1.0, 1.0, 1.0],
        [0.0, 0.0, 1.0, 1.0],
    ],
    control_points=[
        [0.0, 0.0],
        [1.0, 0.0],
        [2.0, 0.0],
        [3.0, 0.0],
        [0.0, 1.0],
        [1.0, 1.0],
        [2.0, 1.0],
        [3.0, 1.0],
    ],
    weights=[1.0, 0.5, 1.0, 1.0, 1, 1, 0.5, 1],
)
nurbs.create.parametric_view(conform=True).show()
```
This will plot like so:
image

Is this what you intended?

File: splinepy/helpme/create.py [hunk 1]
```diff
@@ -295,10 +304,47 @@ def parametric_view(spline, axes=True):
...
+
+        # process knots to insert
+        if spline.has_knot_vectors:
+            for i, (kv, d) in enumerate(
+                zip(spline.knot_vectors, spline.degrees)
+            ):
+                n_repeating = int(d + 1)
+                query = kv[n_repeating:-n_repeating]
+                if len(query) > 0:
+                    para_spline.insert_knots(i, query)
```
Comment (jzwar) [inline splinepy/helpme/create.py:336]: This can be simplified, as starting #214, you can just pass all knots at the same time to simplify the syntax.

File: splinepy/helpme/create.py [hunk 2]
```diff
@@ -295,10 +304,47 @@ def parametric_view(spline, axes=True):
...
+        # conforming weights
+        if spline.is_rational:
+            para_spline.ws[:] = spline.ws
+
+    elif spline.has_knot_vectors:
+        # in non-conform case, we can just add missing unique knots
+        for i, uk in enumerate(spline.unique_knots):
+            query = uk[1:-1]
+            if len(query) > 0:
+                para_spline.insert_knots(i, query)
```
Comment (jzwar) [inline splinepy/helpme/create.py:347]: Same here

**Review by jzwar — APPROVED** [review]
LGTM, thanks! I don't know exactly where this will be required, but feel free to merge...

**Comment by j042** [comment]
> This PR looks reasonable to me for non-rational splines. However I am not sure about the meaningfulness of the parametric view of rational splines. For example, consider the following spline:
> 
> ```
> import splinepy as sp
> import numpy as np
> # Create a nurbs
> nurbs = sp.NURBS(
>     degrees=[2, 1],
>     knot_vectors=[
>         [0.0, 0.0, 0.0, 0.5, 1.0, 1.0, 1.0],
>         [0.0, 0.0, 1.0, 1.0],
>     ],
>     control_points=[
>         [0.0, 0.0],
>         [1.0, 0.0],
>         [2.0, 0.0],
>         [3.0, 0.0],
>         [0.0, 1.0],
>         [1.0, 1.0],
>         [2.0, 1.0],
>         [3.0, 1.0],
>     ],
>     weights=[1.0, 0.5, 1.0, 1.0, 1, 1, 0.5, 1],
> )
> nurbs.create.parametric_view(conform=True).show()
> ```
> 
> This will plot like so: image
> 
> Is this what you intended?

not really, thanks for pointing this out!

**Comment by j042** [comment]
Screenshot 2023-08-23 at 15 13 40

The goal was to visualize boundary conditions in parametric view. but in this case, I see that knot lines are somewhat confusing

File: splinepy/helpme/create.py (hunk 1 above)
Comment (j042) [reply splinepy/helpme/create.py:336]: I don't see any related changed in #214 but you can help me for next pr
